### PR TITLE
WIP Signature test

### PR DESCRIPTION
**Pull request by rlkelly**
## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.

### CODE REVIEW

File: include/TrustWalletCore/TWTezosAddress.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+TW_EXPORT_STATIC_METHOD
+bool TWTezosAddressIsValidString(TWString *_Nonnull string);
+
+/// Initializes an address from a base58 sring representaion.
+TW_EXPORT_STATIC_METHOD
+bool TWTezosAddressInitWithString(struct TWTezosAddress *_Nonnull address, TWString *_Nonnull string);
+
+/// Initializes an address from a public key.
+TW_EXPORT_STATIC_METHOD
+bool TWTezosAddressInitWithPublicKey(struct TWTezosAddress *_Nonnull address, struct TWPublicKey publicKey);
```
Comment (alejandro-isaza) [inline include/TrustWalletCore/TWTezosAddress.h]: This should now be 
```suggestion
bool TWTezosAddressInitWithPublicKey(struct TWTezosAddress *_Nonnull address, struct TWPublicKey *_Nonnull publicKey);
```
make sure you are on latest master

File: src/Base64.h [hunk 2]
```diff
@@ -5,6 +5,7 @@
 // file LICENSE at the root of the source code distribution tree.
 
 #pragma once
+#define BOOST_NO_AUTO_PTR
```
Comment (alejandro-isaza) [inline src/Base64.h]: 🤔

File: src/Tezos/Address.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+#include "Address.h"
+#include "BinaryCoding.h"
+#include "HexCoding.h"
+
+using namespace TW;
+using namespace TW::Tezos;
+
+Address::Address(const std::string& pkey_hash) {
+  public_key_hash = pkey_hash;
```
Comment (alejandro-isaza) [inline src/Tezos/Address.cpp]: please use 4 spaces for indentation

File: src/Tezos/Address.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+namespace TW {
+namespace Tezos {
+
+class Address {
+public:
+    std::string public_key_hash;
+    /// Initializes a Tezos address with a public_key hash.
+    Address(const std::string& string);
+    /// Returns a string representation of the address.
+    std::string forge() const;
```
Comment (alejandro-isaza) [inline src/Tezos/Address.h]: Let's rename to `string()` for consistency

File: src/Tezos/PublicKey.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "BinaryCoding.h"
+#include <string>
+
+namespace TW {
+namespace Tezos {
+
+class PublicKey {
```
Comment (alejandro-isaza) [inline src/Tezos/PublicKey.h]: Let's use the existing PublicKey implementation if possible.

File: src/Tezos/Signer.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+Data Signer::signHexString(const PrivateKey& privateKey, std::string forgedBytes) {
+  auto watermark = "03";
+  auto watermarkedForgedBytesHex = parse_hex(watermark + forgedBytes);
+  auto hash = Hash::blake2b(watermarkedForgedBytesHex, 32);
+
+  ed25519_public_key pk;
+  ed25519_publickey(privateKey.bytes.data(), pk);
+
+  ed25519_signature signature;
+  ed25519_sign(hash.data(), hash.size(), privateKey.bytes.data(), pk, signature);
```
Comment (alejandro-isaza) [inline src/Tezos/Signer.cpp]: Use `privateKey.sign`

File: src/Tezos/Transaction.h [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+#include "Address.h"
+#include "PublicKey.h"
+#include <string>
+#include <variant>
+
+namespace TW {
+namespace Tezos {
+
+enum OperationKind {
```
Comment (alejandro-isaza) [inline src/Tezos/Transaction.h]: Let's use `enum class` and lowercase cases

**Comment by rlkelly** [comment]
sorry, accidently sent this, please reference https://github.com/TrustWallet/wallet-core/pull/107

File: src/Tezos/Signer.cpp (hunk 6 above)
Comment (rlkelly) [reply src/Tezos/Signer.cpp]: This requires ed22519, which the current private key implementation does not use

Comment (alejandro-isaza) [reply src/Tezos/Signer.cpp]: it does now :)

**Comment by keefertaylor** [comment]
Resolving merge conflicts...

File: src/Tezos/Signer.cpp (hunk 6 above)
Comment (rlkelly) [reply src/Tezos/Signer.cpp]: awesome!

**Pull request closed without merging**